presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+
+
+``query.min-expire-age``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``15 m``
+ * **Description:** This property describes time after which the query metadata may be removed from server. If value is low, it's possible that client will not be able to receive information about query completion. The value describes minimum time that must pass to remove query (after it's considered completed) but if there is space available in history queue the query data will be kept longer. The size of history queue is defined by ``query.max-history`` property (``100`` by default).
+
+
+``query.max-concurrent-queries``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** **Deprecated** Describes how many queries be processed simultaneously in single cluster node. It shouldn't be used in new configuration, the ``query.queue-config-file`` can be used instead.
+
+
+.. _query-max-memory:
+
+``query.max-memory``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``20 GB``
+ * **Description:** Serves as default value for ``query_max_memory`` session property. This property also describes strict limit of total memory allocated around the cluster that may be used to process single query. The query is dropped if the limit is reached unless session want to prevent that by setting session property ``resource_overcommit``. The session may also want to decrease system pressure, so it's possible to decrease query memory limit for session by setting ``query_max_memory`` to smaller value. Setting ``query_max_memory`` to higher value then ``query.max-memory`` will not have any effect. This property may be used to ensure that single query cannot use all resources in cluster. The value should be set to be higher than what typical expected query in system will need - that way system will be resistant to SQL bugs that would cause large unwanted computation. Also if rare queries will require more memory, then the ``resource_overcommit`` session property may be used to break the limit. It is important to set this value to higher then default when presto runs complex queries on large datasets.
+
+
+``query.max-memory-per-node``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``JVM max memory`` * ``0.1``
+ * **Description:** The purpose of that is same as of :ref:`query.max-memory<query-max-memory>` but the memory is not counted cluster-wise but node-wise instead.
```

Comment:
This property is critical for skewed data sets, and in general the ration query.max-memory / worker nodes, should not be reduced or regular jobs (and views) may start failing.